llama_index/response_synthesizers/base.py

```diff
@@ -0,0 +1,173 @@
+"""Response builder class.
+
+This class provides general functions for taking in a set of text
+and generating a response.
+
+Will support different modes, from 1) stuffing chunks into prompt,
+2) create and refine separately over each chunk, 3) tree summarization.
+
+"""
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Generator, List, Optional, Sequence, Union
+
+from llama_index.callbacks.schema import CBEventType, EventPayload
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.indices.service_context import ServiceContext
+from llama_index.response.schema import RESPONSE_TYPE, Response, StreamingResponse
+from llama_index.schema import BaseNode, NodeWithScore, MetadataMode
+from llama_index.types import RESPONSE_TEXT_TYPE
+
+logger = logging.getLogger(__name__)
+
+QueryTextType = Union[str, QueryBundle]
+
+
+class BaseSynthesizer(ABC):
+    """Response builder class."""
+
+    def __init__(
+        self,
+        service_context: Optional[ServiceContext] = None,
+        streaming: bool = False,
+    ) -> None:
+        """Init params."""
+        self._service_context = service_context or ServiceContext.from_defaults()
+        self._callback_manager = self._service_context.callback_manager
+        self._streaming = streaming
+
+    @property
+    def service_context(self) -> ServiceContext:
+        return self._service_context
+
+    @abstractmethod
+    def get_response(
+        self,
+        query_str: str,
+        text_chunks: Sequence[str],
+        **response_kwargs: Any,
+    ) -> RESPONSE_TEXT_TYPE:
+        """Get response."""
+        ...
+
+    @abstractmethod
+    async def aget_response(
+        self,
+        query_str: str,
+        text_chunks: Sequence[str],
+        **response_kwargs: Any,
+    ) -> RESPONSE_TEXT_TYPE:
+        """Get response."""
+        ...
+
+    def _log_prompt_and_response(
+        self,
+        formatted_prompt: str,
+        response: RESPONSE_TEXT_TYPE,
+        log_prefix: str = "",
+    ) -> None:
+        """Log prompt and response from LLM."""
+        logger.debug(f"> {log_prefix} prompt template: {formatted_prompt}")
+        self._service_context.llama_logger.add_log(
```

Comment:
this is deprecated right?

Reply:
yea it is. Might need a PR to rip this out completely later

Reply:
ya might as well rip it out for 0.7.0 

What other deprecation we had marked down, we should just take all of them out for 0.7.0 as well.

Reply:
Hmm, this is the only one that comes to mind. We already ripped out the token counter

Reply:
https://github.com/jerryjliu/llama_index/pull/6710